pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -306,6 +329,28 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterProvision) getInstallPod(job *batchv1.Job, pLog log.FieldLogger) (*corev1.Pod, error) {
+	installPod := &corev1.Pod{}
```

Comment:
Remove this variable. Return nil when there is an error rather than returning a zero `Pod`.